compiler-core/src/dependency.rs

```diff
@@ -63,6 +64,65 @@ where
     Ok(packages)
 }
 
+/**
+* Used to compare 2 versions of a package.
+*/
+pub type PackageVersionDiffs = HashMap<String, (Version, Version)>;
+
+fn resolve_major_versions(
+    package_fetcher: Box<dyn PackageFetcher>,
+    versions: PackageVersions,
+) -> PackageVersionDiffs {
+    versions
+        .iter()
+        .filter_map(|(package, version)| {
+            let Ok(hexpackage) = package_fetcher.get_dependencies(package) else {
+                return None;
+            };
+
+            let Some(latest) = &hexpackage
+                .releases
+                .iter()
+                .map(|release| release.version.clone())
```

Comment:
Can take a reference here and avoid cloning

Reply:
Fixed